test/org/zaproxy/zap/extension/pscanrulesBeta/InformationDisclosureDebugErrorsUnitTest.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2016 The ZAP Development Team
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesBeta;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Test;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+
+public class InformationDisclosureDebugErrorsUnitTest extends PassiveScannerTest {
+	private static final String URI = "https://www.example.com/";
+	private static final String defaultErrorMessage = "Internal Server Error";
+	
+	@Override
+	protected InformationDisclosureDebugErrors createScanner() {
+		return new InformationDisclosureDebugErrors();
+	}
+	
+	@Test
+	public void alertsIfDebugErrorsDisclosed() throws HttpMalformedHeaderException {
+		String[] data = new String[] {
+				"Internal Server Error",
+				"There seems to have been a problem with the",
+				"This error page might contain sensitive information because ASP.NET",
+				"PHP Error"
+		};
+		
+		for (int i = 0; i < data.length; i++) {
+			String debugError = data[i];
+			String responseBody = "<html>" + debugError + "</html>"; 
+			
+			HttpMessage msg = new HttpMessage();
+			msg.setRequestHeader("GET " + URI + " HTTP/1.1");		
+			msg.setResponseBody(responseBody);
+			msg.setResponseHeader(
+					"HTTP/1.1 200 OK\r\n" +
+					"Server: Apache-Coyote/1.1\r\n" +
+					"Content-Type: text/html;charset=ISO-8859-1\r\n" +
+					"Content-Length: " + msg.getResponseBody().length() + "\r\n");
+			rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+			
+			assertThat(alertsRaised.size(), equalTo(i + 1));
+			
+			Alert alert = alertsRaised.get(i);
+			assertThat(alert.getMessage().getResponseBody().toString(), equalTo(responseBody));

```

Comment:
This could check that the message of the alert is the same as the one scanned (instead of just checking the body).